pkg/application/inject/fuse/container.go

```diff
@@ -71,3 +69,165 @@ func collectAllContainerNames(pod common.FluidObject) ([]string, error) {
 
 	return allContainerNames, nil
 }
+
+// changeForInitFuse change the original fuse template for init fuse
+func (s *Injector) changeForInitFuse(runtimeInfo base.RuntimeInfoInterface, template *common.FuseInjectionTemplate, pvcName, files string) error {
+	// 1. check if the files string contain shell command
+	if err := checkShellCommand(files); err != nil {
+		return err
+	}
+
+	fuseContainer := template.FuseContainer
+	mountPath := "/" + common.InitPrefix + pvcName
+
+	// 2. add volumemounts of emptyDir uesd in init phase
+	emptyDir := corev1.VolumeMount{
+		Name:      common.InitPrefix + pvcName,
+		MountPath: mountPath,
+	}
+
+	// 3. add volumemounts of copy configmap
+	copyConfigMap := corev1.VolumeMount{
+		Name:      initcopy.CopyVolName,
+		MountPath: initcopy.CopyScriptPath,
+		ReadOnly:  true,
+		SubPath:   initcopy.CopyScriptName,
+	}
+
+	fuseContainer.VolumeMounts = append(fuseContainer.VolumeMounts, emptyDir, copyConfigMap)
+
+	// 4. add volumes of copy configmap
+	var mode int32 = 0755
+	template.VolumesToAdd = []corev1.Volume{
+		{
+			Name: initcopy.CopyVolName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: initcopy.CopyConfigMapName,
+					},
+					DefaultMode: utilpointer.Int32Ptr(mode),
+				},
+			},
+		},
+		{
+			Name: common.InitPrefix + pvcName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+
+	if err := runtimeInfo.GetOrCreateCopyConfigMap(); err != nil {
+		return err
+	}
+
+	// 5. get and set the shell command
+	var mountShell, mountArgs, checkShell, copyShell, umountShell, dsMountPoint string
+
+	if len(fuseContainer.Command) > 1 {
+		mountShell = fuseContainer.Command[len(fuseContainer.Command)-1]
+	} else {
+		mountShell = fuseContainer.Command[0]
+	}
+
+	for _, line := range fuseContainer.Args {
+		mountArgs = mountArgs + " " + line
+	}
+
+	mountShell = "nohup " + mountShell + mountArgs + " & "
+
+	lifycycle := fuseContainer.Lifecycle
+
+	// PostStart is fixed: bash -c time /check-mount.sh <mount-path> <type>  >> /proc/1/fd/1
+	checkShell = lifycycle.PostStart.Exec.Command[2]
+	lifycycle.PreStop.ProtoMessage()
+
+	dsMountPoint = strings.Split(checkShell, " ")[2]
```

Comment:
What does "ds" mean here?

Reply:
Got it now, does it mean "dataset"? I think `fuseMountPoint` is okay, `ds` mostly indicates `DaemonSet` when talking about K8s.